plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java

```diff
@@ -244,6 +244,21 @@ public void testSelectFromYearlyPartitionedTable()
         assertEquals((long) actualValues.getOnlyValue(), 1L);
     }
 
+    @Test(description = "regression test for https://github.com/trinodb/trino/issues/7784")
+    public void testSelectWithSingleQuoteInWhereClause()
+    {
+        String tableName = "test.select_with_single_quote";
+
+        onBigQuery("DROP TABLE IF EXISTS " + tableName);
+        onBigQuery("CREATE TABLE " + tableName + "(col INT64, val STRING)");
+        onBigQuery("INSERT INTO " + tableName + " VALUES (1,'Looney\\'s Lane')");
```

Comment:
Let's use other literal value for clarifying test objective. 
```suggestion
        onBigQuery("INSERT INTO " + tableName + " VALUES (1, 'escape\\'single quote')");
```